### DISCUSSION TITLE
Heltun Thermostat change on HASS

### DISCUSSION THREAD

**Discussion by roblomq** [zwave-js/zwave-js-ui#2661] [Q&A]
I have a Heltun thermostat which works perfectly with zwavejs2mqtt and autodiscovery in Domoticz. THe only minor thing is that the  buttons on the selector switch is incorrect.

The switch shows: 
      "off",
      "heat",
      "dry",
      "auto",
      "cool",
      "fan_only"

But this must be 
      "Man",
      "Com",
      "Dry",
      "Time",
      "Eco",
      "Vac"

The HASS json code is
`
  "type": "climate",
  "object_id": "climate",
  "discovery_payload": {
    "min_temp": 5,
    "max_temp": 40,
    "temp_step": 0.5,
    "modes": [
      "off",
      "heat",
      "dry",
      "auto",
      "cool",
      "fan_only"
    ],
    "mode_state_template": "{{ {0: \"off\", 1: \"heat\", 8: \"dry\", 10: \"auto\", 11: \"cool\", 13: \"fan_only\"}[value_json.value] | default('off') }}",
    "current_temperature_topic": "zwave/Badkamer_thermostaat/sensor_multilevel/endpoint_0/Air_temperature",
    "current_temperature_template": "{{ value_json.value }}",
    "temperature_state_topic": "zwave/Badkamer_thermostaat/thermostat_setpoint/endpoint_0/setpoint/1",
    "temperature_state_template": "{{ value_json.value }}",
    "temperature_command_topic": "zwave/Badkamer_thermostaat/thermostat_setpoint/endpoint_0/setpoint/1/set",
    "mode_state_topic": "zwave/Badkamer_thermostaat/thermostat_mode/endpoint_0/mode",
    "mode_command_topic": "zwave/Badkamer_thermostaat/thermostat_mode/endpoint_0/mode/set",
    "action_topic": "zwave/Badkamer_thermostaat/thermostat_operating_state/endpoint_0/state",
    "action_template": "{{ {0: \"idle\", 1: \"heating\", 2: \"cooling\", 3: \"fan\", 4: \"idle\", 5: \"idle\", 6: \"fan\", 7: \"heating\", 8: \"heating\", 9: \"cooling\", 10: \"heating\", 11: \"heating\"}[value_json.value] | default('idle') }}",
    "temperature_unit": "C",
    "precision": 0.1,
    "device": {
      "identifiers": [
        "zwavejs2mqtt_xxxx_node37"
      ],
      "manufacturer": "HELTUN",
      "model": "Heating Thermostat (HE-HT01)",
      "name": "Badkamer thermostaat",
      "sw_version": "2.2"
    },
    "name": "Badkamer thermostaat - climate",
    "unique_id": "zwavejs2mqtt_xxxx_Node37_climate"
  },
  "values": [
    "49-0-Air temperature",
    "64-0-mode",
    "67-0-setpoint-1",
    "67-0-setpoint-8",
    "67-0-setpoint-10",
    "67-0-setpoint-11",
    "67-0-setpoint-13",
    "66-0-state"
  ],
  "mode_map": {
    "off": 0,
    "heat": 1,
    "dry": 8,
    "auto": 10,
    "cool": 11,
    "fan_only": 13
  },
  "setpoint_topic": {
    "1": "67-0-setpoint-1",
    "8": "67-0-setpoint-8",
    "10": "67-0-setpoint-10",
    "11": "67-0-setpoint-11",
    "13": "67-0-setpoint-13"
  },
  "default_setpoint": "67-0-setpoint-1",
  "action_map": {
    "0": "idle",
    "1": "heating",
    "2": "cooling",
    "3": "fan",
    "4": "idle",
    "5": "idle",
    "6": "fan",
    "7": "heating",
    "8": "heating",
    "9": "cooling",
    "10": "heating",
    "11": "heating"
  },
  "discoveryTopic": "climate/Badkamer_thermostaat/climate/config",
  "persistent": false,
  "ignoreDiscovery": false,
  "id": "climate_climate"
}`

I did some changes and managed to get the values changed but after a reboot it is changed back to the default. Tried several things with store and update but it seems it is restored to default again after a reboot. 
Is there a way to keep this change for myself but maybe it is also possible to change it in the "database" because cooling is impossible with this [HE-HT01](https://www.heltun.com/z-wave-heating-thermostat) heating thermostat. Heltun has also a cooling thermostat [HE-FT01](https://www.heltun.com/z-wave-fan-coil-thermostat) which can cool of course.

**Comment by robertsLando** [comment]
In order to persist changes you have to store the discovery payloads on the node, there is a presist column that shows if them are persisted or not

**Comment by roblomq** [comment]
Thank you for the information, I will try that.
